test/terra/backends/aer_simulator/test_noise.py

```diff
@@ -156,15 +158,35 @@ def test_kraus_gate_noise(self, method, device):
     def test_kraus_gate_noise_on_QFT(self, method, device):
         """Test Kraus noise on a QFT circuit"""
         shots = 10000
-        noise_model = ref_kraus_noise.kraus_gate_error_noise_models_full()
+
+        # Build noise model
+        error1 = noise.amplitude_damping_error(0.2)
+        error2 = error1.tensor(error1)
+        noise_model = noise.NoiseModel()
+        noise_model.add_all_qubit_quantum_error(error1, ['h'])
+        noise_model.add_all_qubit_quantum_error(error2, ['cp', 'swap'])
+
         backend = self.backend(
             method=method, device=device, noise_model=noise_model)
-        circuit = QFT(3).decompose()
-        circuit.measure_all()
-        ref_target = ref_kraus_noise.kraus_gate_error_counts_on_QFT(shots)
-        result = backend.run(circuit, shots=shots).result()
+        ideal_circuit = transpile(QFT(3), backend)
+
+        # manaully build noise circuit
+        noise_circuit = QuantumCircuit(3)
+        for inst, qargs, cargs in ideal_circuit.data:
+            noise_circuit.append(inst, qargs, cargs)
+            if inst.name == "h":
+                noise_circuit.append(error1, qargs)
+            elif inst.name in ["cp", "swap"]:
+                noise_circuit.append(error2, qargs)
+        # compute target counts
+        noise_state = DensityMatrix(noise_circuit)
+        ref_target = {i: shots * p for i, p in noise_state.probabilities_dict().items()}
+
+        # Run sim
+        ideal_circuit.measure_all()
+        result = backend.run(ideal_circuit, shots=shots).result()
         self.assertSuccess(result)
-        self.compare_counts(result, [circuit], [ref_target], delta=0.1 * shots)
+        self.compare_counts(result, [ideal_circuit], [ref_target], hex_counts=False, delta=0.1 * shots)
```

Comment:
I'm in favour of getting rid of the hard-coded counts, but is there another reason for changing the test in this PR?

Reply:
It started failing but on some jobs, and after debugging it seemed that the ref counts were wrong but I couldn't find any reason in Aer for why they would have changed.